tests/RobotFramework/tests/tedge/tedge_upload_cert.robot

```diff
@@ -6,14 +6,40 @@ Resource    ../../resources/common.resource
 Library    ThinEdgeIO
 
 Test Tags    theme:cli    theme:mqtt    theme:c8y
-Suite Setup            Setup
+Suite Setup    Custom Setup
 Suite Teardown         Get Logs
 
 *** Test Cases ***
+Create the certificate    
+    #You can then check the content of that certificate.
+    ${output}=    Execute Command    sudo tedge cert show    #You can then check the content of that certificate.
+    Should Contain    ${output}    Device certificate: /etc/tedge/device-certs/tedge-certificate.pem
+    Should Contain    ${output}    Subject: CN=${DEVICE_SN}, O=Thin Edge, OU=Test Device
+    Should Contain    ${output}    Issuer: CN=${DEVICE_SN}, O=Thin Edge, OU=Test Device
+    Should Contain    ${output}    Valid from:
+    Should Contain    ${output}    Valid up to:
+    Should Contain    ${output}    Thumbprint:
+
+Renew the certificate  
+    ${output}=    Execute Command    sudo tedge cert renew    stderr=${True}    stdout=${False}    ignore_exit_code=${True}    
+    Should Contain    ${output}    Certificate was successfully renewed, for un-interrupted service, the certificate has to be uploaded to the cloud
+    #You can then check the content of that certificate.
+    ${output}=    Execute Command    sudo tedge cert show    #You can then check the content of that certificate.
```

Comment:
This test is easy to fool as a broken renew command that simply emit a message would pass the test.

The test should check that the certificate file has been actually changed.

The test must check that after the certificate uploaded to c8y and and tedge reconnected to c8y, then the device can correctly connect to c8y.